File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int p = part.size();
        int n = s.size();
        if (p > n) return s;
        stack<char> check;
        string temp = "";
        for(int i = 0; i < n; i++){
            temp = "";
            check.push(s[i]);
            if(check.size() >= p){
                for(int j = 0; j < p; j++){
                    temp = check.top() + temp;
                    check.pop();
                }
                if(temp != part){
                    for(int j = 0; j < p; j++){
                        check.push(temp[j]);
                    }
                }
            }
        }
        temp = "";
        while(!check.empty()){
            temp = check.top() + temp;
            check.pop();
        }
        return temp;
    }
};

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        int p = part.length();
        int n = s.length();
        if(p > n) return s;
        else{
            Stack<Character> check = new Stack<>();
            String result = "";
            for(int i = 0; i < n; i++){
                result = "";
                check.push(s.charAt(i));
                if(check.size() >= p){
                    for(int j = 0; j < p; j++){
                        result = check.pop() + result;
                    }
                    if(!result.equals(part)){
                        for(int j = 0; j < p; j++){
                            check.push(result.charAt(j));
                        }
                    }
                }
            }
            result = "";
            while(!check.empty()){
                result = check.pop() + result;
            }
            return result;
        }
    }
}